docs/proposals/scheduling/20220722-multi-hierarchy-elastic-quota-management.md

```diff
@@ -0,0 +1,365 @@
+is arbitrary.
+
+### Implementation Details
+
+#### Calculate RuntimeQuota
+
+We use an example to introduce how to calculate "runtime". Suppose the cluster total resource is 100, and has 4 quotas,
+the configuration and "request" of each quota group described as below:
+
+![image](/docs/images/runtimequota1.jpg)
+
+We first calculate the "min" part of "runtime". It should be like as below:
+
+![image](/docs/images/runtimequota2.jpg)
+
+Then we find quota groupA can lent 5 quotas to B\C\D, and the cluster has 40 quotas to allocate, so the sum is 45 for B\C\D 
+to share. We take the proportion of "max" as the basis for B\C\D to allocate shared resources by default, because the larger 
+the "max" is, the more resources it want. In this way, "max" has both the meaning of resource ceiling and allocation proportion. 
+At the same time, we also support using new fields to represent the allocation proportion, which is called "shareRatio". 
+
+In the above example, we assume that "shareRatio" is equal to "max". According to "max" proportion, quota groupB can get
+45 * 60 / (60 + 50 + 80) = 14. quota group C can get 45 * 50 / (60 + 50 + 80) = 12. quota group D can get 45 * 80 / (60 + 50 + 80) = 19.
+However, quota groupB only need 5 more due to request is 20 and min is 15, and quota group C and D are still hungry, 
+so quota groupB can share 14 - 5 = 9 to C and D.
+
+![image](/docs/images/runtimequota3.jpg)
+
+quota group C and D can still share the remained quota of 9 by allocation proportion, which C get 9 * 50 / (50 + 80) = 3, 
+D get 9 * 80 / (50 + 80) = 6, and we get the runtime of each quota group finally.
+
+![image](/docs/images/runtimequota4.jpg)
+
+The whole process can be summarized as follows:
+
+1. The quota divided into two categories, one is whose "request" is less than "min", we call it "lent-quotas". The other is 
+whose "request" is greater than "min", we call it "borrow-quotas".
+
+2. Calculate the "runtime" of each quota group not exceed "min", so we can get how many resources can be lent and shared to 
+"borrow-quotas".
+
+3. The "borrow-quotas" share the lent resource by allocation proportion. 
+
+4. If the new "runtime" is larger than "request", there will be new resources which can be lent and shared to the rest "borrow-quotas".
+
+It must be pointed out that if the cluster resources suddenly decrease due to node failure, the sum of "min" may be 
+greater than the total resources of the cluster. If this case happens, we can't grantee "min" of each quota group actually. 
+So we will reduce the "min" of each quota group in a moderate proportion, which is to ensure that the sum of 
+"min" actually in effect is less than the total resources of the cluster.
+
+We also need to introduce the concept of "sys-group". "sys-group" means that the "min" of this quota group is infinite, 
+and its request will never be bound by the quota. When the scheduler starts, the quota group 0 will be created by default, 
```

Comment:
> When the scheduler starts, the quota group 0 will be created by default

We can create the `sys-group` quota when installing the koordinator.

Reply:
done